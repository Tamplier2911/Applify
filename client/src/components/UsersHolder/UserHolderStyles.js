import styled from "styled-components";

export const UsersHolderContainer = styled.div``;
