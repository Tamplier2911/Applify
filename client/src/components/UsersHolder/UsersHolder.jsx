// import './usersHolder.scss';
import React, { useEffect, useState } from "react";
import { withRouter } from "react-router-dom";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { loadAllUsersStart } from "../../redux/users/users.actions";
import {
  selectAllUsers,
  selectUsersIsLoading
} from "../../redux/users/users.selectors";

// components
import WithSpinnerHOC from "../WithSpinnerHOC/WithSpinnerHOC";
import CollectionHolder from "../CollectionHolder/CollectionHolder";
import CollectionForHolder from "../CollectionForHolder/CollectionForHolder";
import Button from "../Button/Button";
import UserCard from "../UserCard/UserCard";
import FormInput from "../FormInput/FormInput";

// data transformation
import { simpleUserSearch } from "../../utils/DataTransformations/simpleSearches";

// JS Rendering CSS
import {
  UsersHolderContainer,
  UserHolderSearchBarContainer
} from "./UsersHolderStyles";

// component constants
import usersHolderData from "../../utils/ComponentUsersHolderConstants/componentUsersHolderConstants";

// buff collection for holder with spinner
const CollectionForHolderWithSpinner = WithSpinnerHOC(CollectionForHolder);

const UsersHolder = ({
  lang,
  history,
  loadAllUsersStart,
  users,
  isLoading
}) => {
  useEffect(() => {
    if (!users.length) {
      loadAllUsersStart();
    }
  }, [loadAllUsersStart, users.length]);

  const [searchInput, setSearchInput] = useState({ search: "" });
  const { search } = searchInput;

  const onInputChange = e => {
    const { name, value } = e.target;
    setSearchInput({ ...searchInput, [name]: value });
  };

  const redirectToCreateUser = () => {
    history.push("/profile/users/create");
  };

  const usersArray = simpleUserSearch(users, search);

  const {
    usersHolderTitle,
    usersHolderButton,
    usersHolderSearch
  } = usersHolderData[lang];

  return (
    <UsersHolderContainer>
      <Button
        type="button"
        value={usersHolderButton}
        click={redirectToCreateUser}
      />
      <UserHolderSearchBarContainer>
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={search}
          name="search"
          label={usersHolderSearch}
          type="text"
        />
      </UserHolderSearchBarContainer>
      <CollectionHolder
        lang={lang}
        title={usersHolderTitle}
        refresher={1}
        cb={() => loadAllUsersStart()}
      >
        <CollectionForHolderWithSpinner
          isLoading={isLoading}
          dataCollection={usersArray}
        >
          <UserCard />
        </CollectionForHolderWithSpinner>
      </CollectionHolder>
    </UsersHolderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
  users: selectAllUsers,
  isLoading: selectUsersIsLoading
});

export default withRouter(
  connect(mapStateToProps, { loadAllUsersStart })(UsersHolder)
);
