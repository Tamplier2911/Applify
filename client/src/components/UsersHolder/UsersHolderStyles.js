import styled from "styled-components";

export const UsersHolderContainer = styled.div``;

export const UserHolderSearchBarContainer = styled.div`
  display: grid;
  margin-top: 2rem;
`;
