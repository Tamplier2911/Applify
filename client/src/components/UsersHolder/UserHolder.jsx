// import './usersHolder.scss';
import React, { useEffect } from "react";
import { withRouter } from "react-router-dom";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
// import { loadUsersStart } from "../../redux/users/users.actions";
// import {
//   selectAllLoadedFeeds,
//   selectIsLoading
// } from "../../redux/users/users.selectors";

// components
import WithSpinnerHOC from "../WithSpinnerHOC/WithSpinnerHOC";
import CollectionHolder from "../CollectionHolder/CollectionHolder";
// import CollectionForHolder from
import Button from "../Button/Button";

// JS Rendering CSS
import { UsersHolderContainer } from "./UserHolderStyles";

// component constants
import usersHolderData from "../../utils/ComponentUsersHolderConstants/componentUsersHolderConstants";

const UsersHolder = ({ lang, history }) => {
  useEffect(() => {}, []);

  const redirectToCreateUser = () => {
    history.push("/authorization");
  };

  const { usersHolderTitle, usersHolderButton } = usersHolderData[lang];

  return (
    <UsersHolderContainer>
      <Button
        type="button"
        value={usersHolderButton}
        click={redirectToCreateUser}
      />
      <CollectionHolder
        lang={lang}
        title={usersHolderTitle}
        refresher={1}
        cb={() => {}}
      >
        <div>Users Collection</div>
      </CollectionHolder>
    </UsersHolderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default withRouter(connect(mapStateToProps)(UsersHolder));
