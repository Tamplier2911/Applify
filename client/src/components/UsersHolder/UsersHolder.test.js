import React from "react";
import toJson from "enzyme-to-json";

// component
import UsersHolder from "./UsersHolder";

// utils
import {
  setShallowWRC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<UsersHolder />", () => {
  const initialState = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    users: {
      allUsers: [],
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialState);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowWRC(<UsersHolder store={store} />);
  });

  it("renders without an error", () => {
    const usersHolder = findByTestAttr(wrapper, "users-holder");
    expect(usersHolder).toHaveLength(1);
  });

  it("renders children - ", () => {
    const usersHolderSearch = findByTestAttr(wrapper, "users-holder-search");
    expect(usersHolderSearch).toHaveLength(1);
  });

  it("renders children - ", () => {
    const usersHolderColHolder = findByTestAttr(
      wrapper,
      "users-holder-col-holder"
    );
    expect(usersHolderColHolder).toHaveLength(1);
  });

  it("renders children - ", () => {
    const usersHolderColForHolderSp = findByTestAttr(
      wrapper,
      "users-holder-col-for-holder-sp"
    );
    expect(usersHolderColForHolderSp).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const usersHolder = findByTestAttr(wrapper, "users-holder");
    expect(toJson(usersHolder)).toMatchSnapshot();
  });
});
