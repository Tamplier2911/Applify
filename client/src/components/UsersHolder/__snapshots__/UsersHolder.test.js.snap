// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersHolder /> matches snapshot 1`] = `
<styled.div
  data-test="users-holder"
>
  <Button
    click={[Function]}
    type="button"
    value="Create User"
  />
  <styled.div
    data-test="users-holder-search"
  >
    <FormInput
      label="Search"
      name="search"
      onInputChange={[Function]}
      type="text"
      value=""
    />
  </styled.div>
  <CollectionHolder
    cb={[Function]}
    data-test="users-holder-col-holder"
    lang="eng"
    refresher={1}
    title="Users:"
  >
    <Component
      data-test="users-holder-col-for-holder-sp"
      dataCollection={Array []}
      isLoading={false}
    >
      <Connect(UserCard) />
    </Component>
  </CollectionHolder>
</styled.div>
`;
